WordPress/wordpress-develop
# Backport block editor persisted preferences

## Description
Backports the persisted preferences feature from Gutenberg (introduced in https://github.com/WordPress/gutenberg/pull/39795).

Removes the old block editor preferences persistence system.

## Testing

1. Ensure you have the Gutenberg plugin disabled
2. Open a block editor (it's worth testing each of the post editor, site editor, widgets editor and customize widgets editor).
3. Change some preferences (e.g. toggle Top Toolbar on)
4. Clear your local storage data
5. Reload the page

Expected: Preferences should be retained (e.g. Top Toolbar should still be on) despite local storage being cleared

Trac ticket: https://core.trac.wordpress.org/ticket/56467

Props noisysocks costdev

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
